src/apphosting/config.ts

```diff
@@ -55,12 +56,45 @@ export function yamlPath(cwd: string): string | null {
 }
 
 /** Load apphosting.yaml */
-export function load(yamlPath: string): Config {
+export function load(yamlPath: string): yaml.Document {
```

Comment:
optional nit but these methods all feel like good candidates to wrap in a `yaml.ts` that can be shared across firebase products.

Especially the main logic in getEnv and setEnv could likely be genericized and `get/setEnv` can use those helpers.

Reply:
Generally it's better to build abstraction layers once you actually have a second use for the abstraction. Here, loading a document is 2LOC (can be one). get/setEnv are 10 and 18LOC respectively. I'm frankly not even sure how I'd factor them into a utility that isn't just a more skillful usage of the yaml library itself (e.g. maybe you want me to add a push() method?).

Factoring out YAML wrapper over the library while we're still building an understanding of how to use the library well reeks of premature optimization.